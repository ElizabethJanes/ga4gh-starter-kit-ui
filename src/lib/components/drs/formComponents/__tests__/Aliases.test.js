import React from 'react';
import renderer from 'react-test-renderer';
import Aliases from '../Aliases';

let zeroAliases=[];
let multipleAliases=['alias1', 'alias2', 'alias3'];

test('SHOW <Aliases /> should handle zero aliases', () => {
    const aliases = renderer
    .create(<Aliases readOnly={true} aliases={zeroAliases} />)
    .toJSON();
    expect(aliases).toMatchSnapshot();
});

test('SHOW <Aliases /> should handle multiple aliases', () => {
    const aliases = renderer
    .create(<Aliases readOnly={true} aliases={multipleAliases} />)
    .toJSON();
    expect(aliases).toMatchSnapshot();
});

test('NEW and EDIT <Aliases /> should handle 0 aliases', () => {
    const aliases = renderer
    .create(<Aliases readOnly={false} aliases={zeroAliases} />)
    .toJSON();
    expect(aliases).toMatchSnapshot();
});

test('NEW and EDIT <Aliases /> should handle multiple aliases', () => {
    const aliases = renderer
    .create(<Aliases readOnly={false} aliases={multipleAliases} />)
    .toJSON();
    expect(aliases).toMatchSnapshot();
});