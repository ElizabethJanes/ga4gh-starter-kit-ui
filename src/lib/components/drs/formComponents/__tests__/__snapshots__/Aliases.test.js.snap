// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NEW and EDIT <Aliases /> should handle 0 aliases 1`] = `
<div
  className="MuiFormGroup-root"
>
  <div
    className="MuiBox-root MuiBox-root-10"
  >
    <hr
      className="MuiDivider-root"
    />
  </div>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
  >
    Aliases
  </h6>
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignLeft"
  >
    A list of aliases that can be used to identify the DRS Object by additional names.
  </p>
  <div
    className="MuiFormGroup-root MuiFormGroup-row"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div>
          <div
            className="MuiGrid-root MuiGrid-container"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <button
                aria-describedby={null}
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                title="Add alias"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NEW and EDIT <Aliases /> should handle multiple aliases 1`] = `
<div
  className="MuiFormGroup-root"
>
  <div
    className="MuiBox-root MuiBox-root-11"
  >
    <hr
      className="MuiDivider-root"
    />
  </div>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
  >
    Aliases
  </h6>
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignLeft"
  >
    A list of aliases that can be used to identify the DRS Object by additional names.
  </p>
  <div
    className="MuiFormGroup-root MuiFormGroup-row"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiFormGroup-root MuiFormGroup-row"
        >
          <div
            className="MuiFormControl-root"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
                htmlFor="alias-0"
                id="alias-0-label"
              >
                Alias
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="alias-0"
                  name="alias"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={false}
                  required={false}
                  type="text"
                  value="alias1"
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-5"
                  >
                    <span>
                      Alias
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-12"
          >
            <div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
              >
                <button
                  aria-describedby={null}
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  title="Remove this alias."
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11H7v-2h10v2z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiFormGroup-root MuiFormGroup-row"
        >
          <div
            className="MuiFormControl-root"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
                htmlFor="alias-1"
                id="alias-1-label"
              >
                Alias
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="alias-1"
                  name="alias"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={false}
                  required={false}
                  type="text"
                  value="alias2"
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-5"
                  >
                    <span>
                      Alias
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-13"
          >
            <div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
              >
                <button
                  aria-describedby={null}
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  title="Remove this alias."
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11H7v-2h10v2z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiFormGroup-root MuiFormGroup-row"
        >
          <div
            className="MuiFormControl-root"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
                htmlFor="alias-2"
                id="alias-2-label"
              >
                Alias
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="alias-2"
                  name="alias"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={false}
                  required={false}
                  type="text"
                  value="alias3"
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-5"
                  >
                    <span>
                      Alias
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-14"
          >
            <div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
              >
                <button
                  aria-describedby={null}
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  title="Remove this alias."
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11H7v-2h10v2z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div>
          <div
            className="MuiGrid-root MuiGrid-container"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <button
                aria-describedby={null}
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                title="Add alias"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SHOW <Aliases /> should handle multiple aliases 1`] = `
<div
  className="MuiFormGroup-root"
>
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    <hr
      className="MuiDivider-root"
    />
  </div>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
  >
    Aliases
  </h6>
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignLeft"
  >
    A list of aliases that can be used to identify the DRS Object by additional names.
  </p>
  <div
    className="MuiFormGroup-root MuiFormGroup-row"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiFormGroup-root MuiFormGroup-row"
        >
          <div
            className="MuiFormControl-root"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                data-shrink={true}
                htmlFor="alias-0"
                id="alias-0-label"
              >
                Alias
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="alias-0"
                  name="alias"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={true}
                  required={false}
                  type="text"
                  value="alias1"
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-5 PrivateNotchedOutline-legendNotched-6"
                  >
                    <span>
                      Alias
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-7"
          >
            <div />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiFormGroup-root MuiFormGroup-row"
        >
          <div
            className="MuiFormControl-root"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                data-shrink={true}
                htmlFor="alias-1"
                id="alias-1-label"
              >
                Alias
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="alias-1"
                  name="alias"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={true}
                  required={false}
                  type="text"
                  value="alias2"
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-5 PrivateNotchedOutline-legendNotched-6"
                  >
                    <span>
                      Alias
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-8"
          >
            <div />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiFormGroup-root MuiFormGroup-row"
        >
          <div
            className="MuiFormControl-root"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                data-shrink={true}
                htmlFor="alias-2"
                id="alias-2-label"
              >
                Alias
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="alias-2"
                  name="alias"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={true}
                  required={false}
                  type="text"
                  value="alias3"
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-5 PrivateNotchedOutline-legendNotched-6"
                  >
                    <span>
                      Alias
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-9"
          >
            <div />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;

exports[`SHOW <Aliases /> should handle zero aliases 1`] = `
<div
  className="MuiFormGroup-root"
>
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <hr
      className="MuiDivider-root"
    />
  </div>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
  >
    Aliases
  </h6>
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignLeft"
  >
    A list of aliases that can be used to identify the DRS Object by additional names.
  </p>
  <div
    className="MuiFormGroup-root MuiFormGroup-row"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;
